conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/Endpoint.java

```diff
@@ -16,13 +16,33 @@
 
 package com.palantir.conjure.java.undertow.lib;
 
-/**
- * Creates a {@link Routable} which may be registered with a web server. The server is responsible
- * for providing a {@link HandlerContext} allowing API implementors to register APIs using
- * <code>server.api(MyServiceRoutableFactory.of(myServiceImpl)</code>.
- */
+import com.palantir.tokens.auth.ImmutablesStyle;
+import io.undertow.util.HttpString;
+import java.util.Optional;
+import org.immutables.value.Value;
+
+@Value.Immutable(builder = false)
+@ImmutablesStyle
 public interface Endpoint {
 
-    Routable create(HandlerContext context);
+    @Value.Parameter
+    HttpString method();
+
+    @Value.Parameter
+    String template();
+
+    @Value.Parameter
+    Optional<String> serviceName();
+
+    @Value.Parameter
+    Optional<String> name();
+
+    static Endpoint of(HttpString method, String template) {
+        return ImmutableEndpoint.of(method, template, Optional.empty(), Optional.empty());
+    }
+
+    static Endpoint of(HttpString method, String template, String serviceName, String name) {
+        return ImmutableEndpoint.of(method, template, Optional.of(serviceName), Optional.of(name));
```

Comment:
I'd prefer to remove parameters in favor of using builders to avoid cross version compatibility breaks